modules/oe_translation_local/modules/oe_translation_multivalue/src/Form/LocalTranslationRequestForm.php

```diff
@@ -0,0 +1,144 @@
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Form handler for the translation request entity add/edit forms.
+ */
+class LocalTranslationRequestForm extends LocalTranslationRequestFormOriginal {
+
+  /**
+   * The entity field manager.
+   *
+   * @var \Drupal\Core\Entity\EntityFieldManagerInterface
+   */
+  protected $entityFieldManager;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(EntityRepositoryInterface $entity_repository, EntityTypeBundleInfoInterface $entity_type_bundle_info, TimeInterface $time, EntityTypeManagerInterface $entity_type_manager, TranslationSourceManagerInterface $translation_source_manager, AccountInterface $current_user, EntityFieldManagerInterface $entityFieldManager) {
+    parent::__construct($entity_repository, $entity_type_bundle_info, $time, $entity_type_manager, $translation_source_manager, $current_user);
+    $this->entityFieldManager = $entityFieldManager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('entity.repository'),
+      $container->get('entity_type.bundle.info'),
+      $container->get('datetime.time'),
+      $container->get('entity_type.manager'),
+      $container->get('oe_translation.translation_source_manager'),
+      $container->get('current_user'),
+      $container->get('entity_field.manager')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function getExistingTranslationData(TranslationRequestLocal $translation_request, string $langcode): array {
+    $existing_translation_data = parent::getExistingTranslationData($translation_request, $langcode);
+    if (!$existing_translation_data) {
+      return $existing_translation_data;
+    }
+
+    // Reorder the existing translation data deltas.
+    $data = $translation_request->getData();
+    $this->processTranslationData($data, $existing_translation_data, $translation_request->getContentEntity()->getEntityTypeId());
+
+    return $existing_translation_data;
+
+  }
+
+  /**
+   * Processes the translation data recursively to use the correct delta.
+   *
+   * @param array $data
+   *   The translation data.
+   * @param array $existing_translation_data
+   *   The existing translation data.
+   * @param string $entity_type_id
+   *   The entity type ID.
+   */
+  protected function processTranslationData(array $data, array &$existing_translation_data, string $entity_type_id): void {
+    foreach (Element::children($data) as $key) {
+      if (!isset($existing_translation_data[$key])) {
+        continue;
+      }
+
+      $child = $data[$key];
+      foreach (Element::children($child) as $child_key) {
+        $sub_child = $child[$child_key];
+        if (isset($sub_child['entity'])) {
+          $this->processTranslationData($sub_child['entity'], $existing_translation_data[$key][$child_key]['entity'], $sub_child['entity']['#entity_type']);
```

Comment:
We can have a fatal error if `$existing_translation_data[$key][$child_key]['entity']` is null.
It was reproducible by following steps:

1. Enable oe_translation_local
2. Create a node with node type created in oe_translation_multivalue_test
3. fill in all field with at least 2 items
4. run \Drupal\oe_translation_multivalue\TranslationMultivalueColumnInstaller::installColumn for each field in node type.
5. Try to translate created node (you should see fatal error).

Reply:
@sergepavle I couldn;t reproduce this issue. Can you check again and see if you missed any steps?

Reply:
OK, fixed, this was another issue, failing at another point. I fixed it to prevent the crash but not much more we can do for the fields that are in the middle of a re-ordering process when we do the update..